### PR TITLE
PHOENIX-6596 Schema extraction double quotes expressions, resulting i…

### PR DESCRIPTION
**Pull request by richardantal**
…n un-executabe create statements

Co-authored-by: Istvan Toth <stoty@apache.org>

### CODE REVIEW

File: phoenix-core/src/main/java/org/apache/phoenix/util/SchemaUtil.java [hunk 1]
```diff
@@ -1301,6 +1301,29 @@ public static String formatColumnName(String name) {
         return name;
     }
 
+    public static String formatIndexColumnName(String name) {
+        String[] splittedName = name.split("\\.");
```
Comment (stoty) [inline phoenix-core/src/main/java/org/apache/phoenix/util/SchemaUtil.java:1305]: nit: split is an irregular verb, so this should be splitName

File: phoenix-core/src/main/java/org/apache/phoenix/schema/tool/SchemaExtractionProcessor.java [hunk 2]
```diff
@@ -135,8 +135,8 @@ private String getIndexedColumnsString(PTable indexPTable, PTable dataPTable, St
         StringBuilder indexedColumnsBuilder = new StringBuilder();
 
         for (PColumn indexedColumn : indexPK) {
-            String indexColumn = extractIndexColumn(indexedColumn.getName().getString(), defaultCF);
-            if(indexColumn.equalsIgnoreCase(MetaDataUtil.VIEW_INDEX_ID_COLUMN_NAME)) {
+            String indexColumn = extractIndexColumn(indexedColumn.getExpressionStr(), defaultCF);
```
Comment (stoty) [inline phoenix-core/src/main/java/org/apache/phoenix/schema/tool/SchemaExtractionProcessor.java:138]: so indexedColumn.getExpressionStr() will return the colum name for "simple" indexes ?

File: phoenix-core/src/main/java/org/apache/phoenix/schema/tool/SchemaExtractionProcessor.java [hunk 3]
```diff
@@ -135,8 +135,8 @@ private String getIndexedColumnsString(PTable indexPTable, PTable dataPTable, St
...
+            if(indexColumn == null || indexColumn.equalsIgnoreCase(MetaDataUtil.VIEW_INDEX_ID_COLUMN_NAME)) {
```
Comment (stoty) [inline phoenix-core/src/main/java/org/apache/phoenix/schema/tool/SchemaExtractionProcessor.java:139]: In what case do we get a null  here ?

File: phoenix-core/src/main/java/org/apache/phoenix/schema/tool/SchemaExtractionProcessor.java (hunk 2 above)
Comment (richardantal) [reply phoenix-core/src/main/java/org/apache/phoenix/schema/tool/SchemaExtractionProcessor.java:138]: We still parse it but indexedColumn.getExpressionStr() is better to parse.
indexedColumn.getName().getString() returnes 'QUOTED' for quoted string and ('number' * 'currency') for lowercase columns so we would not be able to differentiate them.
On the other hand  indexedColumn.getExpressionStr() returns these columns as 'QUOTED' and ("number" * "currency")

File: phoenix-core/src/main/java/org/apache/phoenix/schema/tool/SchemaExtractionProcessor.java (hunk 3 above)
Comment (richardantal) [reply phoenix-core/src/main/java/org/apache/phoenix/schema/tool/SchemaExtractionProcessor.java:139]: If indexedColumn.getExpressionStr() is null I return null in extractIndexColumn.

Comment (richardantal) [reply phoenix-core/src/main/java/org/apache/phoenix/schema/tool/SchemaExtractionProcessor.java]: And indexedColumn.getExpressionStr() is null when indexedColumn.getName().getString() was equal to MetaDataUtil.VIEW_INDEX_ID_COLUMN_NAME

**Review by stoty — APPROVED** [review]
+1 LGTM

File: phoenix-core/src/main/java/org/apache/phoenix/schema/tool/SchemaExtractionProcessor.java [hunk 4]
```diff
@@ -184,21 +184,39 @@ private String getIndexedColumnsString(PTable indexPTable, PTable dataPTable, St
...
                 return String.format("%s.%s", schema, name);
             } else {
-                return SchemaUtil.formatColumnName(columnNameSplit[0]);
+                return formatColumnOrExpression(columnNameSplit[0]);
             }
         }
     }
 
+    private String formatColumnOrExpression(String columnOrExpression) {
+        System.out.println("columnOrExpressioncolumnOrExpressioncolumnOrExpression");
```
Comment (gjacoby126) [inline phoenix-core/src/main/java/org/apache/phoenix/schema/tool/SchemaExtractionProcessor.java:206]: Please remove the System.out.printlns

**Review by gjacoby126 — CHANGES_REQUESTED** [review]

**Review by gjacoby126 — APPROVED** [review]
+1, thanks @richardantal

**Pull request merged**